ecosystem-ci.ts

```diff
@@ -23,16 +24,45 @@ cli
 	.option('--commit <commit>', 'vite commit sha to use')
 	.option('--release <version>', 'vite release to use from npm registry')
 	.action(async (suites, options: CommandOptions) => {
+		let poll = false
+		if (
+			options.branch === 'main' &&
+			options.repo === 'vitejs/vite' &&
+			!options.commit
+		) {
+			const res = await fetch(
+				`https://api.github.com/repos/${options.repo}/branches/${options.branch}`,
+			)
+			const {
+				commit: { sha },
+			} = (await res.json()) as { commit: { sha: string } }
+			if (sha) {
+				options.commit = sha
+				actionsCore.setOutput('commit', sha)
+				poll = true
+			}
+		}
 		if (options.commit) {
 			const url = `https://pkg.pr.new/vite@${options.commit}`
-			//eslint-disable-next-line n/no-unsupported-features/node-builtins
-			const { status } = await fetch(url)
-			if (status === 200) {
-				options.release = url
-				delete options.commit
-
-				console.log(`continuous release available on ${url}`)
-			}
+			const maxAttempts = 60 // 5 minutes
```

Comment:
currently vite-ecosystem-ci takes less than one minute to build vite from source. I don't think polling 5 minutes to wait for it to maybe appear on pr.new is the right way to go here.

Either make sure the ecosystem-ci run is only triggered after pr.new is available or skip ahead and build it here.

Reply:
I think it's important to make it consistent across commits, so I'd prefer not to skip ahead. How about throwing an error if it exceeds 1 minutes? The built package should be available at that time.

Reply:
i don't think ecosystem-ci should be reliant on pr.new availability so fallback is needed.

I'd prefer the wait/poll was implemented at the trigger side, not here

Reply:
> I'd prefer the wait/poll was implemented at the trigger side, not here

The trigger is here in this repo.
https://github.com/vitejs/vite-ecosystem-ci/blob/66423f99171509c0d07fa9f5462160c67d0be847/.github/workflows/ecosystem-ci.yml#L15-L17

Reply:
it should still go into that job then but not convinced we need this. what's the advantage of pr.new on scheduled runs?

Reply:
hey! sorry for the long wait, it's been a bit hard on my side. 

it has the same benefits that it has on triggered runs from PRs! it'd avoid the building part on each scheduled run since there's a high chance that commit was already built. 

And if it was not built, we wait a bit until it'd be. 

But anyway, let me know, I'm ok with making either polling or falling back work.

Reply:
building vite here takes 14 seconds (thanks rolldown!)

![image](https://github.com/user-attachments/assets/7c3826ed-d5a5-44ec-88f9-ae6741c7d45a)

is it really going to be faster to check pr.new? Esp. the polling variant feels like overkill. If its there and usable immediately, ok. but else just do it here. scheduled runs (and PR runs too) are unlikely to hit the same commit hash twice, so unless it is cached already there is no benefit for ecosystem-ci

Reply:
I'd like to keep the behavior consistent across the runs, otherwise there'll be problems that are difficult to investigate like https://github.com/vitejs/vite/pull/20080#issuecomment-2899758165.
If the polling is overkill, I think we should always built it here.